optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/pages/spotroster/ShiftBlobPopoverContent.java

```diff
@@ -0,0 +1,217 @@
+
+import static java.lang.Long.parseLong;
+import static java.util.function.Function.identity;
+import static java.util.stream.Collectors.toMap;
+import static org.optaplanner.openshift.employeerostering.gwtui.client.common.FailureShownRestCallback.onSuccess;
+
+@Templated
+public class ShiftBlobPopoverContent implements BlobPopoverContent {
+
+    @Inject
+    @DataField("root")
+    private HTMLDivElement root;
+
+    @Inject
+    @DataField("close-button")
+    private HTMLButtonElement closeButton;
+
+    @Inject
+    @DataField("from-day")
+    private HTMLInputElement fromDay;
+
+    @Inject
+    @DataField("from-hour")
+    private HTMLInputElement fromHour;
+
+    @Inject
+    @DataField("to-day")
+    private HTMLInputElement toDay;
+
+    @Inject
+    @DataField("to-hour")
+    private HTMLInputElement toHour;
+
+    @Inject
+    @DataField("spot")
+    private ListBox spotSelect; //FIXME: Don't use GWT widget
+
+    @Inject
+    @DataField("employee")
+    private ListBox employeeSelect; //FIXME: Don't use GWT widget
+
+    @Inject
+    @DataField("pinned")
+    private HTMLInputElement pinned;
+
+    @Inject
+    @Named("p")
+    @DataField("rotation-employee")
+    private HTMLElement rotationEmployee;
+
+    @Inject
+    @DataField("delete-button")
+    private HTMLButtonElement deleteButton;
+
+    @Inject
+    @DataField("cancel-button")
+    private HTMLButtonElement cancelButton;
+
+    @Inject
+    @DataField("apply-button")
+    private HTMLButtonElement applyButton;
+
+    private BlobPopover popover;
+
+    private ShiftBlobView blobView;
+
+    private Map<Long, Employee> employeesById;
+
+    @Override
+    public void init(final BlobView<?, ?> blobView) {
+
+        this.blobView = (ShiftBlobView) blobView;
+        final ShiftBlob blob = (ShiftBlob) blobView.getBlob();
+        final Shift shift = blob.getShift();
+
+        employeeSelect.clear();
+        employeeSelect.addItem("Unassigned", "-1"); //FIXME: i18n
+
+        //FIXME: Loading the entire list is not a good idea
+        SpotRestServiceBuilder.getSpotList(shift.getTenantId(), onSuccess(spots -> {
+            spots.forEach(s -> this.spotSelect.addItem(s.getName(), s.getId().toString()));
+            spotSelect.setSelectedIndex(spots.indexOf(shift.getSpot()));
+        }));
+
+        //FIXME: Loading the entire list is not a good idea
+        EmployeeRestServiceBuilder.getEmployeeList(shift.getTenantId(), onSuccess(employees -> {
+            this.employeesById = employees.stream().collect(toMap(Employee::getId, identity()));
+            employees.forEach(e -> employeeSelect.addItem(e.getName(), e.getId().toString()));
+            employeeSelect.setSelectedIndex(employees.indexOf(shift.getEmployee()) + 1);
+        }));
+
+        final LocalDateTime start = shift.getTimeSlot().getStartDateTime();
+        fromDay.value = start.getMonth().toString() + " " + start.getDayOfMonth();
```

Comment:
Should do i18n on this

Reply:
There are lots of i18n bits missing.. I figured we could leave this to a later phase. I'll add a FIXME here.